joomla/joomla-cms
# [5.2] Show a warning when there are due tasks

### Summary of Changes

Show next run, show list by next run and show a warning when there are due tasks

### Testing Instructions

- Load the scheduled tasks
- Notice the ordeing by title and the missing Next Run field from the list
- Apply this patch
- Open that page again
- Confrim that the Next Run field is there as well as its orderd by that field so the next task to run will be shown first

![image](https://github.com/joomla/joomla-cms/assets/2596554/11b3fe6d-faba-4353-bf93-c281333329d6)

- Go to the database -> #__scheduler_tasks take one task and set the `next_execution` date to someting in the past
![image](https://github.com/joomla/joomla-cms/assets/2596554/5c5830ea-be83-4310-b48d-ad45f15e6395)

- Reload the scheduled tasks list and notice the message that comes up
![image](https://github.com/joomla/joomla-cms/assets/2596554/bd0cae6b-6d8a-48aa-b342-bf04b838d3c9)

- Use the "Test " button
- Notice the message is gone, it will come up of course as soon as there is another due task which is not run

### Actual result BEFORE applying this Pull Request

No way to detect not run tasks or get to know the next tasks to trigger.

### Expected result AFTER applying this Pull Request

1. Warning when there are due tasks
2. Option to show the next time a task should run
3. Default ordeing to show the next task to run first

### Link to documentations
Please select:
- [x] Documentation link for docs.joomla.org: https://help.joomla.org/proxy?keyref=Help51:Scheduled_Tasks&lang=en
